cmd/zoekt-sourcegraph-indexserver/main.go

```diff
@@ -465,29 +469,74 @@ func (s *Server) Index(args *indexArgs) (state indexState, err error) {
 		return indexStateEmpty, createEmptyShard(args)
 	}
 
+	repositoryName := args.BuildOptions().RepositoryDescription.Name
+	if _, ok := s.deltaBuildRepositoriesAllowList[repositoryName]; ok {
+		repositoryID := args.BuildOptions().RepositoryDescription.ID
+		debug.Printf("Server.Index: marking %q (ID %d) for delta build", repositoryName, repositoryID)
+
+		args.UseDelta = true
+	}
+
 	reason := "forced"
+
 	if args.Incremental {
 		bo := args.BuildOptions()
 		bo.SetDefaults()
 		incrementalState, fn := bo.IndexState()
 		reason = string(incrementalState)
 		metricIndexIncrementalIndexState.WithLabelValues(string(incrementalState)).Inc()
-		switch incrementalState {
-		case build.IndexStateEqual:
-			debug.Printf("%s index already up to date. Shard=%s", args.String(), fn)
-			return indexStateNoop, nil
 
-		case build.IndexStateMeta:
-			log.Printf("updating index.meta %s", args.String())
+		if !args.UseDelta {
```

Comment:
with my advice earlier up about zoekt-git-index being smart enough to fallback to non delta, I don't think this function should change here. we should hide delta as much as possible, and not couple it tightly all over indexserver. In the future the only thing indexserver should care about w.r.t. delta is we should prepare a repo with the older commits.

Reply:
I can see an argument for that. If push everything (all the error handling) to zoekt-git-index though, we're committing to immediately performing a full build if something goes wrong with the delta build (since we can't communicate with the scheduler from zoekt-git-index). I _think_ that sounds alright, but it seems like a non-trivial decision.

Reply:
non-trivial decision in design, or non-trivial decision for git-index to make?

The way I think of this is:
- indexserver is logic for deciding what to index.
- git-index is to do indexing

This decision on what type of build to do /etc feels like it falls in the indexing domain.